Add dedicated secrets file for Reader

mokagio opened the pull request:
As a result, the secrets files for the other apps updated too. That's because since the last update we removed the App Center API key from them (see also
https://github.com/wordpress-mobile/WordPress-iOS/commit/9348997a1dbed073c541ee7ce31bf19fae0d7e06)

Part of https://linear.app/a8c/issue/CMM-260/configure-reader-target

<!-- blue -->
> [!NOTE]  
> The only Reader-specific values currently tracked are the WordPress.com app id and the Sentry DSN

mokagio committed:
Add dedicated secrets file for Reader

As a result, the secrets files for the other apps updated too.
That's because since the last update we removed the App Center API key
from them.

See also
https://github.com/wordpress-mobile/WordPress-iOS/commit/9348997a1dbed073c541ee7ce31bf19fae0d7e06

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/308">25.9</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Set `WP_APP_URL_SCHEME` for Reader

mokagio committed:
Use appropriate `WP_BUILD_CONFIGURATION` for Reader debug and release

mokagio committed:
Make Reader use its own `ApiCredentials` value

mokagio commented on config/Reader.alpha.xcconfig:
```diff
@@ -3,4 +3,4 @@
 WP_PUSH_NOTIFICATION_APP_ID = TBD
 WP_BUILD_CONFIGURATION = alpha
-WP_APP_URL_SCHEME = TBD
+WP_APP_URL_SCHEME = wpreader
```
This value could also be specialized by build configuration, e.g.:

https://github.com/wordpress-mobile/WordPress-iOS/blob/2281871ee128e48c7e6d1781cba16a041d213c5a/config/WordPress.debug.xcconfig#L6

https://github.com/wordpress-mobile/WordPress-iOS/blob/2281871ee128e48c7e6d1781cba16a041d213c5a/config/WordPress.release.xcconfig#L6

but I'm not sure if we need to?

crazytonyli replied:
`WP_APP_URL_SCHEME` should be one for each app identifier. If I'm not mistaken, debug and release builds share the same app identifier, but alpha builds use a different one.

mokagio committed:
Fix `Bundle` for Notifications in Reader Tab VC

We need to use the bundle to the class, otherwise UIKit will look for
the storyboard in the app (Reader) when the file is in Keystone.

mokagio commented on .configure:
```diff
@@ -32,9 +32,12 @@
       "encrypt": true
+    },
+    {
+      "file": "iOS/Reader/Reader-Secrets.swift",
```
Something not obvious to me was that the source file cannot be called `Secrets.swift`, even though the Reader prefix seems redundant in a folder called Reader.

The reason for that is that when `configure` encrypts the files locally, it uses the file name. And given we already have a source file called `Secrets.swift` we cannot name this new one that.

mokagio commented on WordPress/Classes/Apps/Reader/ReaderTabViewController.swift:
```diff
@@ -108,7 +108,7 @@ final class ReaderTabViewController: UITabBarController, UITabBarControllerDeleg
 
     private func makeNotificationsViewController() -> UIViewController {
-        let notificationsVC = UIStoryboard(name: "Notifications", bundle: nil)
+        let notificationsVC = UIStoryboard(name: "Notifications", bundle: Bundle(for: Self.self))
```
Required to avoid a crash due to the storyboard in Reader being part of Keystone, not the app bundle

![image](https://github.com/user-attachments/assets/c70432e8-76d4-4c57-8369-15135bfaf93e)

mokagio replied:
https://github.com/wordpress-mobile/WordPress-iOS/pull/24464 follows up to this and addresses all such accesses for the notifications storyboard.

mokagio committed:
Constrain `NibLoadable` to `AnyObject`

This is so we can make it use the bundle of the class conforming to it
instead of using `.main`, which fails if the class and its nib are in a
framework (Keystone)
